### PR TITLE
[flutter_markdown] Strip leading whitespace from list items

### PR DESCRIPTION
**Pull request by krozett**
In line with standard Markdown practice, whitespace at the beginning of list items is removed.

Fixes [Flutter issue #134847
](https://github.com/flutter/flutter/issues/134847)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by domesticmouse — COMMENTED** [review]
Please add a test to make sure this doesn't revert later

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
PTAL @stuartmorgan

File: packages/flutter_markdown/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.6.18+2
+
+* Removes leading whitspace from list items.
```
Comment (tarrinneal) [inline packages/flutter_markdown/CHANGELOG.md:3]: typo: whitespace

**Review by tarrinneal — APPROVED** [review]
Approval with the typo fix. Thanks for throwing this together.

**Review by tarrinneal — CHANGES_REQUESTED** [review]
typo change then approval :)

File: packages/flutter_markdown/CHANGELOG.md (hunk 1 above)
Comment (krozett) [reply packages/flutter_markdown/CHANGELOG.md]: Fixed!

**Review by tarrinneal — APPROVED** [review]

**Pull request merged**